src/main/java/uk/co/ramp/Outbreak.java

```diff
@@ -82,102 +86,40 @@ private void generateInitialInfection() {
     }
 
 
-
-    private void runToCompletion() {
+    void runToCompletion() {
         int timeLimit = properties.timeLimit();
         int maxContact = contactRecords.keySet().stream().max(Comparator.naturalOrder()).orElseThrow(RuntimeException::new);
         int runTime;
         boolean steadyState = properties.steadyState();
         double randomInfectionRate = diseaseProperties.randomInfectionRate();
 
         if (timeLimit <= maxContact) {
-            LOGGER.info("Not all contact data will be used");
+            LOGGER.warn("Not all contact data will be used");
             runTime = timeLimit;
             steadyState = false;
         } else {
             LOGGER.info("Potential for steady state soln");
             runTime = maxContact;
         }
 
-        boolean complete = runContactData(runTime, population, contactRecords, randomInfectionRate);
+        boolean complete = runContactData(runTime, randomInfectionRate);
 
         if (steadyState && !complete) {
-            runToSteadyState(runTime, timeLimit, population);
-        }
-
-        outputInfectionData(population);
-
-
-    }
-
-    private void outputInfectionData(Map<Integer, Case> population) {
-        List<Case> infections = new ArrayList<>();
-        for (Case c : population.values()) {
-            if (c.status() != SUSCEPTIBLE) {
-                infections.add(c);
-            }
-        }
-
-        infections.sort(Comparator.comparingInt(Case::exposedTime));
-
-        Map<Integer, Set<Integer>> infectors = new HashMap<>();
-
-        for (Case c : infections) {
-            infectors.putIfAbsent(c.exposedBy(), new HashSet<>());
-
-            Set<Integer> var = infectors.get(c.exposedBy());
-
-            var.add(c.id());
-            infectors.put(c.exposedBy(), var);
+            runToSteadyState(runTime, timeLimit);
         }
 
-        // initial infectees
-        Set<Integer> infectees = infectors.get(Case.getInitial());
+        // Output map of infections
+        new InfectionMap(population).outputMap();
```

Comment:
I would suggest the following:
```suggestion
        new InfectionMap(Collections.unmodifiableMap(population)).outputMap();
```
which will provide an unmodifiable view (not a copy) to the InfectionMap() object, to ensure the InfectionMap does not accidentally/unnecessarily amend the contents of our population map

Reply:
Sounds good.